static/app/utils/profiling/spanTree.tsx

```diff
@@ -44,21 +37,24 @@ class SpanTreeNode {
 }
 
 class SpanTree {
-  spans: RawSpanType[];
   spanTree: SpanTreeNode = SpanTreeNode.Root();
   orphanedSpans: RawSpanType[] = [];
 
   constructor(spans: RawSpanType[]) {
-    this.spans = spans;
-    this.buildCollapsedSpanTree();
+    this.buildCollapsedSpanTree(spans);
+
+    this.spanTree.span.timestamp = this.spanTree.children.reduce(
+      (sum, c) => sum + c.span.timestamp - c.span.start_timestamp,
+      0
+    );
```

Comment:
Isn't this computing the sum of the span durations? Why is it being stored on the root span's `timestamp` field?

I think what you want here is to collect the top level `start_timestamp` and `timestamp` from the transaction itself to and copy that to the root node to use as the overall transaction duration.